src/codegen/transformations.lisp

```diff
@@ -123,3 +124,37 @@ both CL namespaces appearing in `node`"
            (type parser:identifier-list bound-variables)
            (values boolean &optional))
   (null (intersection (node-variables node) bound-variables)))
+
+(defun node-returns-p (node)
+  "Does `node` contain a `node-return` subnode?"
+  (declare (type node node)
+           (values boolean &optional))
+  (traverse
+   node
+   (list
+    (action (:before node-return _node)
```

Comment:
I forgot the semantics of return. Does

    (fn () (return Unit))

work by returning from the nearest enclosed lambda?

Reply:
Answer: yes, `return` returns from the lambda, so this PR will need adjustments to account for that